pkg/controller/priorityqueue/priorityqueue.go

```diff
@@ -277,6 +277,13 @@ func (w *priorityqueue[T]) GetWithPriority() (_ T, priority int, shutdown bool)
 	w.waiters.Add(1)
 
 	w.notifyItemOrWaiterAdded()
+
+	// ref: https://github.com/kubernetes-sigs/controller-runtime/issues/3239
```

Comment:
@alvaroaleman Shouldn't we do this at the beginning of the func?

Reply:
https://github.com/kubernetes-sigs/controller-runtime/pull/3250